tests/widgetresize.js

```diff
@@ -0,0 +1,324 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* globals document, Event */
+
+import WidgetResize from '../src/widgetresize';
+
+// ClassicTestEditor can't be used, as it doesn't handle the focus, which is needed to test resizer visual cues.
+import ClassicEditor from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
+import ArticlePluginSet from '@ckeditor/ckeditor5-core/tests/_utils/articlepluginset';
+
+import {
+	toWidget
+} from '../src/utils';
+import {
+	setData as setModelData
+} from '@ckeditor/ckeditor5-engine/src/dev-utils/model';
+
+import Rect from '@ckeditor/ckeditor5-utils/src/dom/rect';
+
+describe( 'WidgetResize', () => {
+	let editor, editorElement, widget, mouseListenerSpies;
+
+	const commitStub = sinon.stub();
+	const mouseMock = {
```

Comment:
I think it's more of a simulator than a mock. It could be a mock if it was replacing something. Here, it's a new thing.